charts/kube-state-metrics/templates/deployment.yaml

```diff
@@ -146,9 +151,14 @@ spec:
       topologySpreadConstraints:
 {{ toYaml .Values.topologySpreadConstraints | indent 8 }}
       {{- end }}
-      {{- if .Values.kubeconfig.enabled}}
+      {{- if or (.Values.kubeconfig.enabled) (.Values.volumes) }}
```

Comment:
I believe this can be omitted

Reply:
Please see my answer above about empty key